internal/request/graphql/schema/collection.go

```diff
@@ -596,6 +622,94 @@ func policyFromAST(directive *ast.Directive) (client.PolicyDescription, error) {
 	return policyDesc, nil
 }
 
+func embeddingFromAST(directive *ast.Directive, fieldDef *ast.FieldDefinition) (client.EmbeddingDescription, error) {
+	embedding := client.EmbeddingDescription{
+		FieldName: fieldDef.Name.Value,
+	}
+	for _, arg := range directive.Arguments {
+		switch arg.Name.Value {
+		case types.EmbeddingDirectivePropFields:
+			val, ok := arg.Value.(*ast.ListValue)
+			if !ok {
```

Comment:
todo: The cast checking (`!ok`) checks in `embeddingFromAST` and `contraintsFromAST` are untested, and it looks like they should be covered by the GQL lib when it parses the SDL.  Please either cover them with integration tests, or remove them.

Reply:
Done